sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -307,6 +308,9 @@ def executable_stage_transform(
           for side in side_inputs
       },
                           main_input=main_input_id)
+      # at this point we should have resolved an environment, as the key of
+      # components.environments cannot be None.
+      assert self.environment is not None
```

Comment:
We discussed this before and said this was safe, but just wanted to double check.

Reply:
Yes, that looks safe.